com.unity.render-pipelines.universal/Editor/ShaderGraph/UniversalProperties.cs

```diff
@@ -0,0 +1,105 @@
+        public static string ZWrite(bool isShaderGraph)                        { return isShaderGraph ? SG_ZWrite : HW_ZWrite; }
+        public static string ZWriteControl(bool isShaderGraph)                 { return isShaderGraph ? SG_ZWriteControl : null; }   // no HW equivalent
+        public static string ZTest(bool isShaderGraph)                         { return isShaderGraph ? SG_ZTest : null; }          // no HW equivalent
+        public static string Cull(bool isShaderGraph)                          { return isShaderGraph ? SG_Cull : HW_Cull; }
+        public static string CastShadows(bool isShaderGraph)                   { return isShaderGraph ? SG_CastShadows : HW_CastShadows; }
+        public static string ReceiveShadows(bool isShaderGraph)                { return isShaderGraph ? SG_ReceiveShadows : HW_ReceiveShadows; }
+        public static string QueueOffset(bool isShaderGraph)                   { return isShaderGraph ? SG_QueueOffset : HW_QueueOffset; }
+
+        // for hand-written shaders (Lit.shader, Unlit.shader, maybe others)
+        public static readonly string HW_SpecularWorkflowMode = "_WorkflowMode";
+        public static readonly string HW_Surface = "_Surface";
+        public static readonly string HW_Blend = "_Blend";
+        public static readonly string HW_AlphaClip = "_AlphaClip";
+        public static readonly string HW_SrcBlend = "_SrcBlend";
+        public static readonly string HW_DstBlend = "_DstBlend";
+        public static readonly string HW_ZWrite = "_ZWrite";
+        public static readonly string HW_Cull = "_Cull";
+        public static readonly string HW_CastShadows = "_CastShadows";
+        public static readonly string HW_ReceiveShadows = "_ReceiveShadows";
+        public static readonly string HW_QueueOffset = "_QueueOffset";
+
+        // for ShaderGraph shaders (renamed more uniquely to avoid potential naming collisions with HDRP properties and user properties)
+        public static readonly string SG_SpecularWorkflowMode = "_URP_WorkflowMode";
+        public static readonly string SG_Surface = "_URP_Surface";
+        public static readonly string SG_Blend = "_URP_Blend";
+        public static readonly string SG_AlphaClip = "_URP_AlphaClip";
+        public static readonly string SG_SrcBlend = "_URP_SrcBlend";
+        public static readonly string SG_DstBlend = "_URP_DstBlend";
+        public static readonly string SG_ZTest = "_URP_ZTest";
+        public static readonly string SG_ZWrite = "_URP_ZWrite";
+        public static readonly string SG_ZWriteControl = "_URP_ZWriteControl";
+        public static readonly string SG_Cull = "_URP_CullMode";
+        public static readonly string SG_CastShadows = "_URP_CastShadows";
+        public static readonly string SG_ReceiveShadows = "_URP_ReceiveShadows";
+        public static readonly string SG_QueueOffset = "_URP_QueueOffset";
```

Comment:
Not having consistency between all URP shaders is a bad decision imo. It also violates the constraint 2 in your document. 
 
 This will not only make it hard for user animating property themselves as it will also break/not work with existing tooling we have in Unity like AnimationClip.  cc @amechtley

Reply:
This PR is intended to not break any existing shader or material behavior, and we're doing extensive testing to ensure that.  

There is an edge case where if the user happens to have added a property on their shader that matches the name of one of the new target-generated properties, the shader will start issuing errors.  This makes it important to use very unique names for these properties.

Andre brought up that this PR does break an existing _workflow_ :   iterating on shader surface settings and viewing the changes in scene.  The global "Allow Material Override" checkbox that I am working on adding will fix that, by making it possible for the user to select whether they want the new Uber-shader behavior or not.   Actually, now that I think about it, since the default value there will be "false" this will also mean they won't even get the edge case I described above unless they check the "Allow Material Override" box.

Reply:
For the reasons described in the document, it is not possible to satisfy all of the constraints simultaneously.  Option 1 is selected as the least-bad solution.

Reply:
How often do users animating surface settings like alpha clipping, blend mode, transparent/opaque?
These are the only properties that will be made unique.

Reply:
I haven't pushed the unique keyword change yet, but the set that would be made unique are the ones driven by the properties, that can be set independently in different render pipelines:
_ALPHATEST_ON
_SURFACE_TYPE_TRANSPARENT
_ALPHAPREMULTIPLY_ON
_ALPHAMODULATE_ON

The following are using shared state between pipelines, so I think they can be kept as shared names (though it's a bit dangerous as any discrepancy between HDRP and URP setting of these may cause bugs):
_NORMALMAP
_EMISSION